### ISSUE TITLE
Compatibility with Sphinx 5.0+

### ISSUE THREAD

**Issue by emezh** [speedyleion/sphinx-c-autodoc#17]
During sphinx clean-up for 5.0 (specifically this change: https://github.com/sphinx-doc/sphinx/commit/267954edba498dcc132f98c6a6e7fd029e8f6bbb#diff-f2b81eb446ad45461b255667a70a375c8548595c28846185fef603078fc8ab96), the 2nd parameter in prepare_docstring(...) in sphinx/util/docstrings.py was removed.
Consequently, sphinx-c-autodoc throws an exception when calling  prepare_docstring:

Traceback (most recent call last):
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/cmd/build.py", line 281, in build_main
    app.build(args.force_all, args.filenames)
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/application.py", line 347, in build
    self.builder.build_update()
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 310, in build_update
    self.build(to_build,
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 326, in build
    updated_docnames = set(self.read())
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 433, in read
    self._read_serial(docnames)
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 454, in _read_serial
    self.read_doc(docname)
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/builders/__init__.py", line 510, in read_doc
    publisher.publish()
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/core.py", line 217, in publish
    self.document = self.reader.read(self.source, self.parser,
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/io.py", line 104, in read
    self.parse()
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/readers/__init__.py", line 78, in parse
    self.parser.parse(self.input, document)
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/parsers.py", line 78, in parse
    self.statemachine.run(inputlines, document, inliner=self.inliner)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 170, in run
    results = StateMachineWS.run(self, input_lines, input_offset,
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2769, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 393, in new_subsection
    newabsoffset = self.nested_parse(
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 281, in nested_parse
    state_machine.run(block, input_offset, memo=self.memo,
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/statemachine.py", line 239, in run
    context, next_state, result = self.check_line(
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/statemachine.py", line 451, in check_line
    return method(match, context, next_state)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2342, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2354, in explicit_construct
    return method(self, expmatch)
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2096, in directive
    return self.run_directive(
  File "/home/me/venv_test/lib/python3.8/site-packages/docutils/parsers/rst/states.py", line 2146, in run_directive
    result = directive_instance.run()
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/ext/autodoc/directive.py", line 148, in run
    documenter.generate(more_content=self.content)
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx/ext/autodoc/__init__.py", line 918, in generate
    docstrings: List[str] = sum(self.get_doc() or [], [])
  File "/home/me/venv_test/lib/python3.8/site-packages/sphinx_c_autodoc/__init__.py", line 293, in get_doc
    return [prepare_docstring(docstring, ignore, tab_width)]
TypeError: prepare_docstring() takes from 1 to 2 positional arguments but 3 were given

**Comment by speedyleion** [comment]
Thanks for reporting the issue, and the detail.
Due to personal time constraints I won't be able to look into this for a few weeks.

**Comment by speedyleion** [comment]
Ok finally got time to start looking at this.
While I know the main issue is that this extension doesn't work with sphinx 5.0. I want to better automate being informed of these breaking changes instead of leaving it to consumers to report. 
So:
- I just merged in #18 which pins everything back to sphinx 3.1 this is to provide a passing CI.
- I next will work on getting dependabot going, with everything pinned back it should create PRs to update.
  - I pulled in poetry to leverage it's lock file with dependabot, but I still plan on using the setup.py to do package build/release. The reason is to leverage the hard caps from poetry and dependabot, while leaving `setup.py` with looser version requirements for consumers. I'm up for suggestions on better ways to handle this.
- After dependabot I will work on building up a sphinx CI matrix, thinking `3.1` and `4.0` and when dependabot says to update sphinx to `6.0` will add `5.0` to the matrix
- Then I want to get mypy and pylint passing again on ci
- Finally I will try to come up with a deprecation plan.  Currently thinking 3 years which would mean sphinx 3.x support would be dropped June 2023.

Thanks again for reporting this, providing a suggested patch, and your patience as I try to get through this.

**Comment by speedyleion** [comment]
doh, I forgot it would close the issue when merged...
The mainline works with sphinx 5, but I'm hoping to get a release out by end of week, more or less finishing #20

**Comment by speedyleion** [comment]
@emezh Version 1.1.0 is now release, https://pypi.org/project/sphinx-c-autodoc/1.1.0/
let me know if there are any issues

Thanks!

**Comment by emezh** [comment]
Thanks for the fix.
I've created a fresh venv and installed the latest sphinx and sphinx-c-autodoc (it pulled sphinx-5.3.0 and sphinx-c-autodoc-1.1.0)
HTML generation looks ok on a quick inspection (no warnings/errors), but rst generation is not: `sphinx-c-apidoc` cmd line utility is not created/installed under `bin/`.

Also, you may want to add `sphinx_c_autodoc.__version__` variable so it is easy to check the installed version in run-time if needed (up to you though).

p.s. i'll be offline till January.

**Comment by speedyleion** [comment]
Thanks for the feed back.  Looks like the way I got dependabot stuff going, I messed up the packaging such that it used the pinned dependabot one.  I've since released https://pypi.org/project/sphinx-c-autodoc/1.1.1/ which should now include the CLI `sphinx-c-autodoc` and the looser Sphinx, Clang, BeautifulSoup4 dependency versions
Since I wanted to minimize the change in 1.1.1 to the fix.  I created #52 to address the `__version__` suggestion.
